components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationUserResidentResolverServiceImpl.java

```diff
@@ -225,6 +226,25 @@ the accessed organization (lower level)
         return basicOrganizationList;
     }
 
+    public User getUserFromResidentOrgId(String userId, String userResidentOrganizationId)
+            throws OrganizationManagementException {
+
+        User user = null;
```

Comment:
Without throwing null in case when user is not found, we can return Optional<User>.

Reply:
What's the advantage of returning Optional instead of null here? feel like it will degrade the performance.

Reply:
https://www.developer.com/java/java-optional-object/#:~:text=Programmers%20can%20also%20use%20the,using%20the%20empty()%20method.

Reply:
https://www.developer.com/java/java-optional-object/#:~:text=For%20one%2C%20it%20can%20add%20significant%20overhead%20to%20code%20execution%20time%2C%20as%20the%20Optional%20wrapper%20must%20be%20created%20and%20checked%20each%20time%20a%20variable%20is%20accessed.

Reply:
Yes. there are pros and cons in these type of workarounds. Ideal way is to do null check before consuming in order to prevent NPE. But developers missed it and hence we try to use these workarounds. Please check the similar methods https://github.com/wso2-extensions/identity-organization-management-core/blob/main/components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationUserResidentResolverService.java#L42~L54

Reply:
Resolves [c2d07e3](https://github.com/wso2-extensions/identity-organization-management-core/pull/61/commits/c2d07e393f1c829bf025c808cc53edec5deab1e1)